src/vanilla/Model/CompositeTypeTS.cs

```diff
@@ -33,6 +33,35 @@ public override Property Add(Property item)
             return result;
         }
 
+        private List<CompositeTypeTS> polymorphicUnionCases;
+
+        /// <summary>Contains the immediate polymorphic children of this class.</summary>
+        public List<CompositeTypeTS> PolymorphicUnionCases
+        {
+            get
+            {
+                if (polymorphicUnionCases == null)
+                {
+                    if (BaseIsPolymorphic)
+                    {
+                        polymorphicUnionCases =
+                            CodeModel.ModelTypes
+                                .Where(m => m.BaseModelType == this)
+                                .Cast<CompositeTypeTS>()
+                                .ToList();
+                    }
+                    else
+                    {
+                        polymorphicUnionCases = new List<CompositeTypeTS>() { };
+                    }
+                }
+                return polymorphicUnionCases;
+            }
+        }
+
+        /// <summary>The name to use when referring to this type in a parameter, return type or property definition.</summary>
+        public string ReferenceName => PolymorphicUnionCases.Any() ? Name + "Union" : Name.ToString();
```

Comment:
Maybe call this `UnionTypeName`? It's basically saying that if this type is part of union, then use this type's name with the "Union" suffix. Otherwise, just use this type's name.

Reply:
Might make sense. I thought the name conveyed the purpose: when referencing this type in an annotation, use this ReferenceName, and when declaring it or referencing it in a mapper, use Name. But it's a little contrived no matter what you do.